File: csharp/Leet/Old/HouseRobber.cs
// https://leetcode.com/problems/house-robber/description/
// NeetCode solution: https://www.youtube.com/watch?v=73r3KWiEvyk

namespace Old.HouseRobber
{
    public class Solution
    {
        public int Rob(int[] nums)
        {
            int rob1 = 0, rob2 = 0;
            foreach (int n in nums)
            {
                var temp = Math.Max(n + rob1, rob2);
                rob1 = rob2;
                rob2 = temp;
            }
            return rob2;
        }

    }
}

File: returned/python/today15/house_robber.py
# https://leetcode.com/problems/house-robber/description/
from typing import List

class Solution:
    def rob(self, nums: List[int]) -> int:
        rob1, rob2 = 0, 0

        for n in nums:
            tmp = max(rob1 + n, rob2)
            rob1 = rob2
            rob2 = tmp

        return rob2